hedyorg/hedy
# 💻 Preselect language filter based on user language

This PR makes the following changes:
- when a user opens the public adventures page, the user language is used to preset the language filter, so that we show language-relevant adventures. Note that if there are no adventures for the language of the user, no language filter is set, so that we display some adventures rather than no adventures at all.
- public adventures which do not have a language property set used to be displayed as if they are in the language of the current user. Now this is changed, so that if a language filter is applied, only adventures with specified language are displayed. Language-less adventures appear only when there is no language filter applied. 

Fixes #5368

**How to test**
Ensure that you have public adventures for at least 2 languages (e.g. English and Dutch) and also adventures that do not have language set.
- If the user's preferred language is set to Dutch, ensure that opening the public adventures page will have the Dutch language filter pre-selected and only Dutch adventures are displayed. If the user changes the language filter to English, then only English adventures should be displayed. If the user clears the language filter, then the all adventures should be displayed: English, Dutch and the ones without a language.
- If the user's preferred language is set to anything other than English or Dutch, opening the public adventures page should not preset a language filter. Thus, all adventures should be displayed: English, Dutch and language-less.
- Editing and cloning adventures should work. If changed or cloned adventure does not have language set, the language of the user should be set as a property of the adventure.

Please note that clearing the filter on tags is not possible right now. This seems like a bug which is not in the scope of the current task and perhaps should be addressed in the future. However, it will impact your testing.
